agents/redfish/fence_redfish.py

```diff
@@ -159,7 +161,16 @@ def main():
         else:
             options["--systems-uri"] = sysresult["uri"]
 
-    result = fence_action(None, options, set_power_status, get_power_status, None)
+    reboot_fn = None
+    if options["--action"] == "diag":
+        # Diag is a special action that can't be verified so we will reuse reboot functionality
+        # to minimize impact on generic library
+        options["original-action"] = options["--action"]
+        options["--action"] = "reboot"
```

Comment:
Doesn't this defeat the purpose?

`diag` is supposed to send an NMI, causing the node to panic and dump a vmcore. If we set `options["--action"] = "reboot"` before calling `fence_action()`, I believe this will power the node off and on. That could interrupt the vmcore collection.

Reply:
That's why it has the return-part to not do the rest of it.

Reply:
Based on fence_ipmilan implementation btw.

Reply:
I didn't notice that we were setting a `reboot_cycle_fn`, so I was following the wrong branch. This makes sense now and looks reasonable.

(I'd like to think I would have caught the fact that my original commit created a no-op for the diag action, if I'd had an environment to test in ;) )

Reply:
It wasnt very obvious that it needed this extra logic, so we only realized it wasnt working when testing it.